cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/AbstractSparkSubmitter.java

```diff
@@ -145,6 +145,13 @@ protected abstract void addMaster(Map<String, String> configs, ImmutableList.Bui
     return Collections.emptyList();
   }
 
+  /**
+   * Returns true if spark driver has succeeded.
+   */
+  protected boolean hasSparkDriverSucceeded() throws Exception {
```

Comment:
Why not define this as an abstract method?

Reply:
the reason is currently there are multiple(3) implementations of this abstract class. however we need this method to be only implemented in `MasterEnvironmentSparkSubmitter`. 
Added default implementation so that we dont have to add implementations in other implementation classes.